### PR TITLE
Resolves #1648: Logging when unloading the page

### LINKED ISSUE #1648: Ajax Synchronous Request Failing on page unload events in Chrome
As @aileenzeng originally identified here, she came across the following [bug](https://github.com/ProjectSidewalk/SidewalkWebpage/pull/1645#issuecomment-483932975):

> "NetworkError: Failed to execute 'send' on 'XMLHttpRequest': Failed to load 'http://0.0.0.0:9000/task': Synchronous XHR in page dismissal."
> 
> From this [StackOverflow post](https://stackoverflow.com/questions/55676319/ajax-synchronous-request-failing-in-chrome), it looks like it's because chrome doesn't allow synchronous ajax requests anymore. Apparently using [a beacon](https://developer.mozilla.org/en-US/docs/Web/API/Beacon_API/Using_the_Beacon_API) is the alternative others are using, but it looks like it's still an experimental feature.

We need to review all places that we use a synchronous AJAX call and begin coming up with alternatives.

### PR DESCRIPTION
**Pull request by andrew4699**
Resolves #1648 

Added two endpoints ```/taskBeacon``` and ```/validationTaskBeacon``` that accept plaintext JSON and simulate a redirect to ```/task``` and ```/validationTask``` respectively with the plaintext converted to "actual" JSON format.

The beacons on the client side now use these endpoints since ```navigator.sendBeacon``` doesn't support ```application/json```.

To test:
First try to reproduce this on develop:
* comment out @misaugstad 's temporary fix in "public/..../SVLabel/..../Form.js" since that's what we're trying to replace. The line to comment out is the self.submitData line after the "TODO: Replace with beacon (or some ajax alternative) asap. Starter code below" comment
* make sure that "Unload" events (and other events close to a page refresh) aren't being recorded in the **audit_task_interaction** and **validation_task_interaction** tables

Now test on this branch:
- Check that "Unload" events are being recorded in the **audit_task_interaction** and **validation_task_interaction** tables
- Check that labels/validations done right before a refresh save

### CODE REVIEW

File: public/javascripts/SVLabel/src/SVLabel/data/Form.js [hunk 1]
```diff
@@ -17,7 +17,8 @@
 function Form (labelContainer, missionModel, missionContainer, navigationModel, neighborhoodModel, panoramaContainer, taskContainer, mapService, compass, tracker, params) {
     var self = this;
     var properties = {
-        dataStoreUrl : undefined
+        dataStoreUrl : undefined,
+        beaconDataStoreUrl : undefined,
```
Comment (misaugstad) [inline public/javascripts/SVLabel/src/SVLabel/data/Form.js]: This shouldn't have a comma

File: public/javascripts/SVLabel/src/SVLabel/data/Form.js [hunk 2]
```diff
@@ -276,24 +278,19 @@ function Form (labelContainer, missionModel, missionContainer, navigationModel,
             asyncParam = false;
 
         // Old code: this does not work on the newest versions of Google Chrome.
```
Comment (misaugstad) [inline public/javascripts/SVLabel/src/SVLabel/data/Form.js]: You can remove this line. Also remove the `asyncParam` above along with that big paragraph I wrote.

File: public/javascripts/SVLabel/src/SVLabel/data/Form.js [hunk 3]
```diff
@@ -276,24 +278,19 @@ function Form (labelContainer, missionModel, missionContainer, navigationModel,
...
+        var data = [self.compileSubmissionData(task)];
         var jsonData = JSON.stringify(data);
-        var headers = {
-            type: 'application/x-www-form-urlencoded'
-        };
-
-        var blob = new Blob([jsonData], headers);
-        navigator.sendBeacon(properties.dataStoreUrl, blob);
+        navigator.sendBeacon(properties.beaconDataStoreUrl, jsonData);
+        console.log(properties.beaconDataStoreUrl, properties, "jsonData", jsonData);
```
Comment (misaugstad) [inline public/javascripts/SVLabel/src/SVLabel/data/Form.js]: Remove this log statement.

File: public/javascripts/SVValidate/src/data/Form.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
-function Form(url) {
+function Form(url, beaconUrl) {
     var properties = {
-        dataStoreUrl : url
+        dataStoreUrl : url,
+        beaconDataStoreUrl : beaconUrl,
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/data/Form.js]: don't need a comma on this line.

File: public/javascripts/SVValidate/src/data/Form.js [hunk 5]
```diff
@@ -104,31 +105,18 @@ function Form(url) {
         else
             asyncParam = false;
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/data/Form.js]: You can remove the `asyncParm` variable and the paragraph above this.

File: app/controllers/TaskController.scala [hunk 6]
```diff
@@ -166,199 +166,223 @@ class TaskController @Inject() (implicit val env: Environment[User, SessionAuthe
...
+    )
+  }
+
   /**
    * Parse the submitted data and insert them into tables.
    *
    * @return
    */
   def post = UserAwareAction.async(BodyParsers.parse.json) { implicit request =>
+    println("posted " + (System.currentTimeMillis / 1000))
```
Comment (misaugstad) [inline app/controllers/TaskController.scala]: you can remove this print statement.

**Comment by andrew4699** [comment]
Just resolved all the comments.

**Pull request merged**